linkerd/app/core/src/errors.rs

```diff
@@ -171,33 +181,36 @@ impl<RspB: Default + hyper::body::HttpBody> respond::Respond<http::Response<RspB
             Err(error) => {
                 warn!("Failed to proxy request: {}", error);
 
-                if let Respond::Http2 { is_grpc } = self {
+                if self.version == http::Version::HTTP_2 {
                     if let Some(reset) = error.h2_reason() {
                         debug!(%reset, "Propagating HTTP2 reset");
                         return Err(error);
                     }
+                }
 
-                    if *is_grpc {
-                        let mut rsp = http::Response::builder()
-                            .version(http::Version::HTTP_2)
-                            .header(http::header::CONTENT_LENGTH, "0")
-                            .body(ResponseBody::default())
-                            .expect("app::errors response is valid");
-                        let code = set_grpc_status(&*error, rsp.headers_mut());
-                        debug!(?code, "Handling error with gRPC status");
-                        return Ok(rsp);
+                // Gracefully teardown the serverside connection.
+                if should_teardown_connection(&*error) {
+                    if let Some(c) = self.close.as_ref() {
+                        debug!("Closing serverside connection");
```

Comment:
very nit: server-side or server side? Only noticed because I first saw this debug message when testing and it caught my attention.